solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -905,7 +906,61 @@ protected boolean addFL(StringBuilder fl, String field, boolean additionalAdded)
     return true;
   }
 
+  protected abstract static class ShardDocQueue {
```

Comment:
If you do an interface, you could have `ShardFieldSortedHitQueue` implement that interface.  Then, `DefaultShardDocQueue` could be removed.

Reply:
I like that idea, thanks for the suggestion.

Though on trying it out, then it didn't quite work out since `ShardFieldSortedHitQueue` would need to implement the `push` as either `add` or `insertWithOverflow` and making that choice doesn't seem particularly intuitive: https://github.com/apache/lucene/blob/releases/lucene/10.2.2/lucene/core/src/java/org/apache/lucene/util/PriorityQueue.java#L150-L184

However, making `DefaultShardDocQueue` into an anonymous class seems a good alternative, both locally for `QueryComponent` and if a deriving class needed to maintain a container alongside of `ShardFieldSortedHitQueue` or instead of it.